### PR TITLE
NEXUS-21171 - Add API endpoints for creating R repositories

### PR DESCRIPTION
**Pull request by aornatovskyy**
**Ticket:** https://issues.sonatype.org/browse/NEXUS-21171

### CODE REVIEW

File: nexus-repository-r-it/src/test/java/org/sonatype/nexus/plugins/r/internal/ScriptResourceIT.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+        nexusFeature("org.sonatype.nexus.plugins", "nexus-repository-r")
+    );
+  }
+
+  @Before
+  public void setup() throws Exception {
+    BaseUrlHolder.set(this.nexusUrl.toString());
+
+    final CloseableHttpClient build = clientBuilder().build();
+    Client client2 = restClientFactory.create(
```
Comment (chernikovmaksym) [inline nexus-repository-r-it/src/test/java/org/sonatype/nexus/plugins/r/internal/ScriptResourceIT.java]: I can see no more clients :D

Comment (aornatovskyy) [reply nexus-repository-r-it/src/test/java/org/sonatype/nexus/plugins/r/internal/ScriptResourceIT.java]: baeae96a23d0b67b148a7ebf576185d2c08cf2c6

File: nexus-repository-r-it/src/test/java/org/sonatype/nexus/plugins/r/internal/ScriptResourceIT.java [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    Client client = restClientFactory.create(
+        RestClientConfiguration.DEFAULTS.withHttpClient(() -> build));
+    client.register(new BasicAuthentication("admin", "admin123"));
+    scriptClient = restClientFactory.proxy(ScriptClient.class, client,
+        UriBuilder.fromUri(nexusUrl.toURI()).path("service/rest").build());
+  }
+
+  @Test
+  public void createRProxyScript() {
+    String repoName = "r-script-repo";
```
Comment (sergeyonofreychuck) [inline nexus-repository-r-it/src/test/java/org/sonatype/nexus/plugins/r/internal/ScriptResourceIT.java]: rename to scriptName

Comment (sergeyonofreychuck) [reply nexus-repository-r-it/src/test/java/org/sonatype/nexus/plugins/r/internal/ScriptResourceIT.java]: Introduce separate variables for repo name and script name

Comment (aornatovskyy) [reply nexus-repository-r-it/src/test/java/org/sonatype/nexus/plugins/r/internal/ScriptResourceIT.java]: b03ffabce361829783b883a427076c91b548fdc6

File: nexus-repository-r/src/main/java/org/sonatype/nexus/repository/r/internal/provisioning/RRepositoryApi.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+ */
+package org.sonatype.nexus.repository.r.internal.provisioning;
+
+import java.util.List;
+
+import org.sonatype.nexus.common.script.ScriptApi;
+import org.sonatype.nexus.repository.Repository;
+import org.sonatype.nexus.repository.r.internal.RFormat;
+import org.sonatype.nexus.repository.storage.WritePolicy;
+
```
Comment (sergeyonofreychuck) [inline nexus-repository-r/src/main/java/org/sonatype/nexus/repository/r/internal/provisioning/RRepositoryApi.java]: since version

**Comment by aornatovskyy** [comment]
After discussion with @wwannemacher  it was decided to not include this functionality

**Pull request closed without merging**